internal/testing/e2e/gateway.go

```diff
@@ -51,10 +52,15 @@ func (p *gatewayTestEnvironment) run(ctx context.Context, namespace string, cfg
 	}
 	k8sSecretClient.AddToMultiClient(secretClient)
 
```

Comment:
You'll also want to change the initialization of the `secretClient` to leverage the scoped client